compiler/rustc_middle/src/ty/adt.rs

```diff
@@ -249,15 +249,25 @@ impl Into<DataTypeKind> for AdtKind {
 }
 
 impl AdtDefData {
+    pub(super) fn new_from_flags(
```

Comment:
this ctor is kinda redundant, i could also just make all the fields `pub(super)` -- *OR* just pull `TyCtxt::mk_adt_def_from_flags`/`TyCtxt::mk_adt_def` into a new impl in this module. I don't really care.